Examples/Mocks/Info.md

```diff
@@ -0,0 +1,41 @@
+## AutoMockable
+
+_Contributed by [@marinbenc](http://twitter.com/marinbenc)_
+
+### For each protocol implementing `AutoMockable` it will...
+Create a class called `ProtocolNameMock` in which it will...
+
+**For each function:**
+ - Implement the function
+ - Add a `functionCalled` boolean to check if the function was called
+ - Add a `functionRecievedArguments` tuple to check the arguments that were passed to the function
+ - Add a `functionReturnValue` variable and return it when the function is called.
+
+**For each variable:**
+ - Add a gettable and settable variable with the same name and type
+
+#### Issues and limitations:
+* Overloaded methods will produce compiler erros since the variables above the functions have the same name. Workaround: delete the variables on top of one of the functions, or rename them.
```

Comment:
he means overloads as in the same name of function so symbols would be duplicated